lib/spack/spack/spec.py

```diff
@@ -4418,7 +4420,10 @@ def old_format(self, format_string="$_$@$%@+$+$=", **kwargs):
         TODO: allow, e.g., ``$6#`` to customize short hash length
         TODO: allow, e.g., ``$//`` for full hash.
         """
-
+        tty.warn(
+            "Using the old Spec.format method."
+            " This method was deprecated in Spack v0.15 and will be removed in Spack v0.20"
```

Comment:
I would also like to suggest an easy way to point the users to where the offending format was read into Spack, but I can't come up with anything. The best I could do myself is to run with:
```console
python -W error bin/spack -d install zlib
```
after I modified the code to use `warnings`. This get us a stacktrace, but is certainly not very "ergonomic".

Maybe we can suggest users to check their configurations and point somewhere in the docs where we show what to use instead?

Reply:
Link to the docs (took some time to find): https://spack.readthedocs.io/en/latest/spack.html#spack.spec.Spec.format